### ISSUE TITLE
PEARC18 Tutorial!

### ISSUE THREAD

**Issue by choldgraf** [jupyterhub/zero-to-jupyterhub-k8s#643]
Hey all, our PEARC tutorial got accepted! Congrats everybody! We can use this issue to coordinate putting together materials and make sure that we're on the right track.

**Conference date**: July 22-26

## What are we doing?
A 6-hour tutorial on JupyterHub and BinderHub

We'll cover:

* Deploying a bare JupyterHub on a commercial cloud platform
* Best-practices in debugging, customizing, and managing a kubernetes deployment
* Customizing the environment for Jupyter with a Docker image
* Make it possible to build environments with a BinderHub
* Demonstrate how this could be done with XSEDE (not live, probably just showing)
* Discuss where the technology is headed more generally.

From our proposal: 
> This day-long workshop will cover setting up BinderHub and JupyterHub in the cloud. By the end, you will have your own public-facing BinderHub deployment that allows users at your own institution to share computational environments with a public link.
> 
> We will begin by deploying a bare JupyterHub on a commercial cloud platform using Kubernetes, a popular technology for robustly managing cloud resources. We'll cover best-practices in debugging, managing, and customizing your Kubernetes deployment, complete with hands-on exercises. We’ll show how to customize the environment your JupyterHub provides to users by using a Docker image of your choice. Next, we’ll cover how to set up a BinderHub service that allows users to automatically create their own computational environments. . We will also deploy the same infrastructure on XSEDE resources to demonstrate how easily this can be generalized to other cloud platforms. 
>
> We’ll close the workshop with a discussion about future directions of the technology and the community, as well as an open question: How can we federate BinderHub instances with support at national and institutional levels?

cc @aculich @jzf2101 @cmd-ntrf @henchc @mpacer @willingc @yuvipanda @minrk

**Comment by jzf2101** [comment]
@choldgraf are we mentioning federation in this?

**Comment by betatim** [comment]
Is there a decision on the topic of federation?

**Comment by choldgraf** [comment]
Nope, I believe that @Carreau 's PR on federation is yet to be merged, so I'm hesitant to commit to one federation model over another until we've got a little bit of experience with an N>2 setup. What do you all think?

**Comment by jzf2101** [comment]
I meant in the presentation

**Comment by choldgraf** [comment]
I think we can decide this closer to the date, depending on whether we have an idea for what the model would be. Otherwise we can mention it in the abstract but not go into a ton of detail.

**Comment by jzf2101** [comment]
@willingc and @yuvipanda have good slides to use as a reference

**Comment by jzf2101** [comment]
Any news on this @aculich ? Can we use our recent scipy submission to help?

**Comment by choldgraf** [comment]
@jzf2101 we did get this accepted! Though there hasn't been much movement on developing new materials (partially since all the attention was taken up by scipy stuff). I think it's time we started working on this though!

**Comment by choldgraf** [comment]
I am not positive I'm going or not...will try and decide on this soon, just figuring out my schedule w/ weddings and such.

How many people do we *need* to be there in order to effectively run the workshop? 2? @aculich do you have an idea for how big these workshops are?

**Comment by aculich** [comment]
it would be great to have at least one person from the Jupyter team be there if possible! I will be there the whole day... and so if we have a total of at least 4 people co-teaching/co-facilitating that would be able to get us through a whole day....  @willingc and/or @yuvipanda able to make it? I think Felix from Compute Canada will be there and I am hoping to sync up with him this week or next week.

**Comment by choldgraf** [comment]
I believe that @yuvipanda is planning to be there though will be a somewhat game-time decision on the day of.

I'm planning to be there too though may just be there for a couple of days rather than the whole conference. @aculich do you think there'd be value in attending the first day or two of PEARC? I looked at the schedule but don't see when the JupyterHub tutorial is.

Also, @aculich or @yuvipanda have you been in touch with [the folks who are presenting this poster](https://ssl.linklings.net/conferences/pearc/pearc18_program/views/includes/files/pos149s2-file2.pdf)? They're presenting on "using JupyterHub on kubernetes with jetstream" which seems relevant :-)

**Comment by willingc** [comment]
Sorry @aculich, I'm booked for the time of PEARC.

**Comment by aculich** [comment]
@choldgraf @yuvipanda @jzf2101 I will let you know as soon as I hear back from the PEARC18 Chairs about registering. I requested a non-attendee presenter code for you to register. I will ping them again if I don't hear back soon. @willingc thanks for letting me know that you won't be able to attend in person— and for all who can't make it to the conference itself, your contributions to the overall efforts are very much appreciated! There's always a lot of work and effort that goes on behind the scenes to make the day-of the event happen and go smoothly. Thanks and I'll keep you all posted!

**Comment by choldgraf** [comment]
Hey all - I updated the z2jh tutorial in the [tutorial folder](https://drive.google.com/open?id=1c48vgJSAmwWyw7KlNYZOBFSDfYObZrK9). I still need to run through the instructions to double-check that it works, but it's now updated with the latest z2jh information.

That said, do we still plan on going through the full kubernetes explanation? @aculich @jzf2101 what do you think? (as opposed to focusing on the littlest jupyterhub)

**Comment by aculich** [comment]
@choldgraf yes, the k8s explanation is important— it is one of the key things that people in this audience need to learn about, probably moreso than the littlelest jhub for this particular audience.
